pkg/operator/certrotationcontroller/satokensigner_controller.go

```diff
@@ -188,11 +190,20 @@ func (c *SATokenSignerController) syncWorker() error {
 		return err
 	}
 
+	needNewSATokenSigningKey := false
 	saTokenSigner, err := c.secretClient.Secrets(operatorclient.OperatorNamespace).Get(c.ctx, "next-service-account-private-key", metav1.GetOptions{})
-	if err != nil && !errors.IsNotFound(err) {
+	if errors.IsNotFound(err) {
+		needNewSATokenSigningKey = true
+	} else if err != nil {
 		return err
+	} else {
+		validKeyPair, err := isValidKeyPair(saTokenSigner.Data["service-account.pub"], saTokenSigner.Data["service-account.key"])
+		if err != nil || !validKeyPair {
+			klog.Errorf("key pair is invalid: %v", err)
```

Comment:
This printing nil made me realize the `isValidKeyPair` has overlapping states. Would it be better to change it into `func validateKeyPair(pubKeyData, privKeyData []byte) (error)` and translate the DeepEgual mismatch into an error?

Reply:
updated!